### PR TITLE
ALFREDOPS-838 First call to Alfred Telemetry endpoint Solr always failing

### PR DESCRIPTION
**Pull request by hechmi-dammak-xenit**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,11 @@ Version template:
  -->
  
 # Alfred Telemetry Changelog
+## [0.9.2] - 2022-12-23
+
+### Fixed
+* First call to Alfred Telemetry endpoint Solr always failing,Fix SolrSnapShoot Metrics [#140]
```
Comment (thijslemmens) [inline CHANGELOG.md:27]: Isn't it SolrSnapShot?

Comment (thijslemmens) [reply CHANGELOG.md:27]: Actually it should be SolrSnapshot

File: alfred-telemetry-solr/alfred-telemetry-solr6/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrSnapshootMetrics.java [hunk 2]
```diff
@@ -4,95 +4,92 @@
...
+import org.apache.solr.response.SolrQueryResponse;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Map;
+import java.util.TimeZone;
+
 public class SolrSnapshootMetrics implements MeterBinder {
```
Comment (thijslemmens) [inline alfred-telemetry-solr/alfred-telemetry-solr6/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrSnapshootMetrics.java:25]: We need to correct the name to SolrSnapshotMetrics

File: alfred-telemetry-solr/alfred-telemetry-solr6/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrSnapshootMetrics.java [hunk 3]
```diff
@@ -4,95 +4,92 @@
...
+    private static final Map<String, Long> statusMapping = Map.of("success", 1L, "failed", 0L);
     private SimpleDateFormat formatter = new SimpleDateFormat("EEE MMM dd HH:mm:ss 'UTC' yyyy");
 
     public SolrSnapshootMetrics(AlfrescoCoreAdminHandler coreAdminHandler) {
         this.coreAdminHandler = coreAdminHandler;
         formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
     }
 
-    private void registerSnapshootMetrics() {
+    private void registerSnapshootMetrics() throws Exception {
```
Comment (thijslemmens) [inline alfred-telemetry-solr/alfred-telemetry-solr6/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrSnapshootMetrics.java:42]: Don't just throw a generic Exception. 
https://stackify.com/common-mistakes-handling-java-exception/

**Review by thijslemmens — CHANGES_REQUESTED** [review]

File: alfred-telemetry-solr/alfred-telemetry-solr6/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrSnapshootMetrics.java (hunk 3 above)
Comment (thijslemmens) [reply alfred-telemetry-solr/alfred-telemetry-solr6/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrSnapshootMetrics.java:42]: So this boils down to Solr code throwing Exception

File: CHANGELOG.md (hunk 1 above)
Comment (thijslemmens) [reply CHANGELOG.md:27]: Nevermind, this naming is all over the place, even in Solr

**Review by thijslemmens — APPROVED** [review]

File: alfred-telemetry-solr/alfred-telemetry-solr6/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrSnapshootMetrics.java (hunk 2 above)
Comment (todorinskiz) [reply alfred-telemetry-solr/alfred-telemetry-solr6/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrSnapshootMetrics.java]: Nope, the terminology in similar classes refers to it as a `snapshoot`, which means the action of taking a snapshot.
https://github.com/apache/solr/blob/26195c82493422cb9d6d4bdf9d4452046e7b3f67/solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java#L231

File: CHANGELOG.md [hunk 4]
```diff
@@ -21,6 +21,11 @@ Version template:
  -->
  
 # Alfred Telemetry Changelog
+## [0.9.2] - 2022-12-23
+
+### Fixed
+* First call to Alfred Telemetry endpoint Solr always failing,Fix SolrSnapShot Metrics [#140]
```
Comment (todorinskiz) [inline CHANGELOG.md:27]: Fix what exactly? The next person reading this should now there was an incompatibility with the implementations we got, namely `AlfrescoReplicationHandler` removal and reference to that doc https://github.com/Alfresco/SearchServices/blob/master/search-services/alfresco-search/doc/architecture/solr-content-store-removal/00001-solr-content-store-removal.md

**Comment by todorinskiz** [comment]
Changelog message is not very clear... Fix what exactly? The next person reading this should now there was an incompatibility with the implementations we got, namely `AlfrescoReplicationHandler` removal and reference to that doc https://github.com/Alfresco/SearchServices/blob/master/search-services/alfresco-search/doc/architecture/solr-content-store-removal/00001-solr-content-store-removal.md

**Pull request merged**